modules/core/src/utils/tesselator.ts

```diff
@@ -169,28 +213,36 @@ export default class Tesselator {
     }
     if (this.normalize || !geometryBuffer) {
       this._forEachGeometry(
-        (geometry, dataIndex) => {
-          geometry = this.normalizeGeometry(geometry);
-          normalizedData[dataIndex] = geometry;
-          vertexStarts[dataIndex + 1] = vertexStarts[dataIndex] + this.getGeometrySize(geometry);
+        (geometry: GeometryT | null, dataIndex: number) => {
+          const normalizedGeometry = geometry && this.normalizeGeometry(geometry);
+          normalizedData[dataIndex] = normalizedGeometry;
+          vertexStarts[dataIndex + 1] =
+            vertexStarts[dataIndex] +
+            (normalizedGeometry ? this.getGeometrySize(normalizedGeometry) : 0);
         },
         startRow,
         endRow
       );
       // count instances
       instanceCount = vertexStarts[vertexStarts.length - 1];
-    } else if (geometryBuffer.buffer instanceof Buffer) {
```

Comment:
Again, using `isBinaryAttribute()` type guard (to be written) would make it more readable and help saving original, less nested control flow.